lighthouse-viewer/test/viewer-test-pptr.js

```diff
@@ -59,80 +67,227 @@ describe('Lighthouse Viewer', function() {
+    const goodPsiResponse = {
+      status: 200,
+      contentType: 'application/json',
+      body: JSON.stringify({lighthouseResult: sampleLhrJson}),
+    };
+    const badPsiResponse = {
+      status: 500,
+      contentType: 'application/json',
+      body: JSON.stringify({error: {message: 'Test error'}}),
+    };
+
+    // Sniffs just the request made to the PSI API. All other requests
+    // fall through.
+    // To set the mocked PSI response, assign `psiResponse`.
+    // To read the intercepted request, use `interceptedRequest`.
+    function onRequest(request) {
+      if (request.url().includes('https://www.googleapis.com')) {
+        interceptedRequest = request;
+        request.respond(psiResponse);
+      } else {
+        request.continue();
       }
-      expected = expected.map(audit => audit.id);
-      const elementIds = await getAuditElementsIds({category, selector: selectors.audits});
+    }
+
+    beforeAll(async () => {
+      await viewerPage.setRequestInterception(true);
+      viewerPage.on('request', onRequest);
+    });
+
+    afterAll(async () => {
+      viewerPage.off('request', onRequest);
+      await viewerPage.setRequestInterception(false);
+    });
+
+    beforeEach(() => {
+      interceptedRequest = undefined;
+      psiResponse = undefined;
+    });
+
+    it('should call out to PSI with all categories by default', async () => {
+      psiResponse = goodPsiResponse;
+
+      const url = `${viewerUrl}?psiurl=https://www.example.com`;
+      await viewerPage.goto(url);
+
+      // Wait for report to render.
+      await viewerPage.waitForSelector('.lh-columns');
+
+      const interceptedUrl = new URL(interceptedRequest.url());
+      expect(interceptedUrl.origin + interceptedUrl.pathname)
+        .toEqual('https://www.googleapis.com/pagespeedonline/v5/runPagespeed');
+
+      // Could use `Object.fromEntries` when we move minimal Node version to 12.
+      const params = {
+        key: interceptedUrl.searchParams.get('key'),
+        url: interceptedUrl.searchParams.get('url'),
+        category: interceptedUrl.searchParams.getAll('category'),
+        strategy: interceptedUrl.searchParams.get('strategy'),
+        locale: interceptedUrl.searchParams.get('locale'),
+        utm_source: interceptedUrl.searchParams.get('utm_source'),
+      };
+      expect(params).toEqual({
+        key: 'AIzaSyAjcDRNN9CX9dCazhqI4lGR7yyQbkd_oYE',
+        url: 'https://www.example.com',
+        // Order in the api call is important to PSI!
+        category: [
+          'performance',
+          'accessibility',
+          'seo',
+          'best-practices',
+          'pwa',
+        ],
+        strategy: 'mobile',
+        // These values aren't set by default.
+        locale: null,
+        utm_source: null,
+      });
+
+      // Confirm that all default categories are used.
+      const defaultCategories = Config.getCategories(defaultConfig).map(c => c.id).sort();
```

Comment:
nit: `Object.keys(defaultConfig.categories).sort()` might be easier than pulling in `Config` and still having to get out the ids

Reply:
> nit: `Object.keys(defaultConfig.categories).sort()` might be easier than pulling in `Config` and still having to get out the ids

and that would allow #9193 to kill `Config.getCategories` which surprisingly returns `Array<{id: string, title: string}>`...it's just an extension-ism pretending to be a general purpose function :)

Reply:
done!